docs/metasploit-framework.wiki/Attacking-ADCS-ESC-Vulnerabilities.md

```diff
@@ -0,0 +1,931 @@
+certificate templates supply either the extended key usage value `Any Purpose EKU`, which
+allows you to use the generate certificate for any purpose, or they don't supply any EKU
+properties at all, which has the same effect, but also allows you to sign new
+certificates.
+
+In Metasploit we take advantage of this ability to request a certificate which can be used
+for any purpose to request a certificate that can be used to authenticate to the domain on
+behalf of another user, thereby giving us permission to access the domain as a more
+privileged user. These attacks therefore require two steps: one to get the certificate
+that gives us permission to issue certificates on behalf of another user, and another
+to get the certificate that actually allows us to authenticate on behalf of a given user,
+and escalate our privileges.
+
+# Finding Vulnerable ESC Templates Using ldap_esc_vulnerable_cert_finder
+Before one can exploit vulnerable ESC templates to elevate privileges, it is necessary to first find a list of vulnerable templates that exist on a domain.
+To do this we can run the `auxiliary/gather/ldap_esc_vulnerable_cert_finder` module. This module will connect to the LDAP server on a target
+Domain Controller (DC), and will run a set of LDAP queries to gather a list of certificate authority (CA) servers and the vulnerable certificate
+templates they make available for enrollment. It will then also query the permissions on both the CA and the certificate template to figure out
+which users or groups can use that certificate template to elevate their privileges.
+
+Keep in mind though that there are two sets of permissions in play here though. There is one set of permissions on the CA server that control
+who is able to enroll in any certificate template from that server, and second set of permissions that control who is allowed to enroll in
+a specific certificate template, which is applied to the certificate template itself. Therefore, the module will also specify which users are
+allowed to enroll in a specific template on a specific CA server, in order to make it as clear as possible which users or groups one needs
+to have access to in order to exploit the vulnerable certificate template.
+
+The following diagram showcases how this permissions check works in a more visual manner:
+
+```mermaid
+flowchart TD
+    user[User] --> firstcheck{CA Server Allows Enrollment?}
+	firstcheck{CA Server Allows Enrollment?} -- YES --> secondcheck{Certificate Template Allows Enrollment?}
+	firstcheck{CA Server Allows Enrollment?} -- NO --> denied[Access Denied]
+	secondcheck{Certificate Template Allows Enrollment?} -- NO --> denied[Access Denied]
+	secondcheck{Certificate Template Allows Enrollment?} -- YES --> success[Access Granted!]
+```
+
+To run the module, you will need to have some the login credentials of a domain joined user. The specific permissions of this user should not matter though,
```

Comment:
```suggestion
To run the module, you will need to have the login credentials of a domain joined user. The specific permissions of this user should not matter though,
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17399/commits/239bc42f4c4228a21dd726e08fa3273545112f7f